src/azure-cli/azure/cli/command_modules/hdinsight/custom.py

```diff
@@ -197,6 +197,16 @@ def create_cluster(cmd, client, cluster_name, resource_group_name, cluster_type,
         )
     ]
 
+    # call get default vm size api to set vm size if customer does not provide the value
+    if not (workernode_size and headnode_size):
+        headnode_size, workernode_size = set_vm_size(cmd.cli_ctx, location, cluster_type, headnode_size,
+                                                     workernode_size)
+
+    if not headnode_size:
+        raise CLIError('Please specify --headnode-size explicitly.')
```

Comment:
Use error type in azclierror file, for example,
https://github.com/Azure/azure-cli/blob/3bff786b9e337aa35bb4cc64a09e5600a35663f8/src/azure-cli-core/azure/cli/core/azclierror.py#L142_L144

Reply:
> Use error type in azclierror file, for example,
> https://github.com/Azure/azure-cli/blob/3bff786b9e337aa35bb4cc64a09e5600a35663f8/src/azure-cli-core/azure/cli/core/azclierror.py#L142_L144

Fixed. Thanks.